### ISSUE TITLE
Unexpected Error during Large uploads.

### ISSUE THREAD

**Issue by kuroderuta** [7x11x13/youtube-up#7]
Trying to upload a file that has 116GB I'm encountering the following error after a while:

(Caused by SSLError(SSLEOFError(8, 'EOF occurred in violation of protocol (_ssl.c:2427)'))), this also invalidates my cookie file so I need to export it again.

I have tried with smaller video files and it works just fine.

**Comment by 7x11x13** [comment]
Have you tried uploading it multiple times? Does it always fail at around the same time?

**Comment by kuroderuta** [comment]
Yes, I tried several times. It fails seemingly towards the end of the upload, maybe even at the very end, calculating from the average disk usage during the upload.

**Comment by 7x11x13** [comment]
Are you able to upload the video to YouTube normally (through the website)?

**Comment by kuroderuta** [comment]
Yes.

**Comment by kuroderuta** [comment]
I implemented the progress (it starts from 20% on upload for some reason) and I always get an error on 70%

```
Step: upload_video, Progress: 70.00%
Authentication error. Retrying in 60 seconds...
Step: start, Progress: 0.00%
Unexpected error during upload: Could not log in to YouTube account. Try getting new cookies
Error type: YTUploaderException
Error details: ('Could not log in to YouTube account. Try getting new cookies',)
```

**Comment by 7x11x13** [comment]
After the large upload fails, are you able to upload a shorter video using the same cookies? Maybe the cookies are becoming invalid before the large file finishes uploading

**Comment by kuroderuta** [comment]
I cannot, a failed upload invalidates the cookies. Seemingly the cookies 'expire' during the large uploads for some reason.

**Comment by 7x11x13** [comment]
I've made the uploader reload cookies after it uploads the file (and before generating session token), so hopefully it's at least possible to upload such long videos if you save new cookies right before the file upload finishes (which is 70% on the progress bar). I have some ideas for a real fix but it's kind of annoying to test this issue...

**Comment by kuroderuta** [comment]
Yeah I understand, I have a fairly fast connection and its still tiresome to test. Anyway I gave the new update a shot, unfortunately still the same behavior as before. Also tested again with a much smaller file with no issues.

**Comment by kuroderuta** [comment]
Hey, any progress on the fix? I tried some of the newer build but the same issue remains. Let me know if you want me to test any builds with large file uploads. Thanks.
